app/services/subscriptions/create_service.rb

```diff
@@ -54,7 +54,7 @@ def process_create
       return result.not_found_failure!(resource: 'plan') unless current_plan
 
       if currency_missmatch?(current_customer&.active_subscription&.plan, current_plan)
-        return result.fail!(code: 'currencies_does_not_match', message: 'currencies does not match')
+        return result.single_validation_failure!(field: :currency, error_code: 'currencies_does_not_match')
```

Comment:
The check to active_subscription and currency will be changed with the customer currency feature.